File: 01_projects/code/java-playwright/cbaps-dawms/src/test/java/com/dawms/pages/ReviewerAssignmentPage.java
package com.dawms.pages;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import com.microsoft.playwright.Locator;
import com.microsoft.playwright.Page;
import com.dawms.library.PlaywrightManager;

/**
 * ReviewerAssignmentPage: Assign reviewers to drug submissions
 * Handles reviewer role selection, reviewer assignment, and routing to signature
 */
public class ReviewerAssignmentPage {

	private static final Logger log = LoggerFactory.getLogger(ReviewerAssignmentPage.class);
	private final Page page;
	private final PlaywrightManager pwm;

	private final Locator reviewerRoleDropdown;
	private final Locator reviewerNameInput;
	private final Locator assignReviewerButton;
	private final Locator continueToSignatureButton;

	public ReviewerAssignmentPage(Page page, PlaywrightManager pwm) {
		this.page = page;
		this.pwm = pwm;

		// Initialize locators
		this.reviewerRoleDropdown = page.locator("#reviewerRole");
		this.reviewerNameInput = page.locator("#reviewerName");
		this.assignReviewerButton = page.locator("button:has-text('Assign')");
		this.continueToSignatureButton = page.locator("button:has-text('Route to Signature')");

		// Stability anchor: wait for page to load
		pwm.waitVisible("text=Reviewer Assignment");
		log.info("Reviewer Assignment Page loaded successfully.");
	}

	/**
	 * Assign a reviewer with specified role and name
	 * @param role - Reviewer role (Clinical Reviewer, Pharmacologist, etc.)
	 * @param reviewerName - Name of reviewer
	 * @return this (for method chaining)
	 */
	public ReviewerAssignmentPage assignReviewer(String role, String reviewerName) {
		reviewerRoleDropdown.selectOption(role);
		log.info("Selected reviewer role: " + role);
		
		pwm.type(reviewerNameInput, reviewerName);
		log.info("Entered reviewer name: " + reviewerName);
		
		pwm.click(assignReviewerButton);
		log.info("Clicked 'Assign' button.");
		
		return this;
	}

	/**
	 * Continue to Signature Routing step
	 * @return SignatureRoutingPage
	 */
	public SignatureRoutingPage routeToSignatureStep() {
		pwm.click(continueToSignatureButton);
		log.info("Clicked 'Route to Signature' button.");
		return new SignatureRoutingPage(page, pwm);
	}
}

File: 01_projects/code/python-playwright/cbaps-dawms/basic-pom/dawms/pages/reviewer_assignment_page.py
"""
ReviewerAssignmentPage: Assign reviewers to drug submissions
Handles reviewer role selection, reviewer assignment, and routing to signature
"""

import logging
from playwright.sync_api import Page, Locator
from dawms.library.playwright_manager import PlaywrightManager

logger = logging.getLogger(__name__)


class ReviewerAssignmentPage:
    """
    ReviewerAssignmentPage handles reviewer assignment workflow.
    This is the Python equivalent of the Java ReviewerAssignmentPage.
    """
    
    def __init__(self, page: Page, pwm: PlaywrightManager):
        """
        Initialize ReviewerAssignmentPage
        
        Args:
            page: Playwright Page object
            pwm: PlaywrightManager instance
        """
        self.page = page
        self.pwm = pwm
        
        # Locators
        self.reviewer_role_dropdown: Locator = page.locator("#reviewerRole")
        self.reviewer_name_input: Locator = page.locator("#reviewerName")
        self.assign_reviewer_button: Locator = page.locator("button:has-text('Assign')")
        self.continue_to_signature_button: Locator = page.locator(
            "button:has-text('Route to Signature')"
        )
        
        # Stability anchor: wait for page to load
        pwm.wait_visible("text=Reviewer Assignment")
        logger.info("Reviewer Assignment Page loaded successfully.")
    
    def assign_reviewer(
        self, role: str, reviewer_name: str
    ) -> 'ReviewerAssignmentPage':
        """
        Assign a reviewer with specified role and name
        
        Args:
            role: Reviewer role (Clinical Reviewer, Pharmacologist, etc.)
            reviewer_name: Name of reviewer
            
        Returns:
            self for method chaining
        """
        self.reviewer_role_dropdown.select_option(role)
        logger.info(f"Selected reviewer role: {role}")
        
        self.pwm.type(self.reviewer_name_input, reviewer_name)
        logger.info(f"Entered reviewer name: {reviewer_name}")
        
        self.pwm.click(self.assign_reviewer_button)
        logger.info("Clicked 'Assign' button.")
        
        return self
    
    def route_to_signature_step(self) -> 'SignatureRoutingPage':
        """
        Continue to Signature Routing step
        
        Returns:
            SignatureRoutingPage instance
        """
        from dawms.pages.signature_routing_page import SignatureRoutingPage
        
        self.pwm.click(self.continue_to_signature_button)
        logger.info("Clicked 'Route to Signature' button.")
        return SignatureRoutingPage(self.page, self.pwm)
